src/code/z_play.c

```diff
@@ -1,81 +1,1608 @@
+                    g = (pixel >> 6) & 0x1F;
+                    b = (pixel >> 1) & 0x1F;
+                    lower = ((r * 2 + g * 4 + b) * 0xF) / 217;
+
+                    *(destI4++) = (upper << 4) | lower;
+                }
+            }
+            break;
+        }
+
+        case 5: {
+            u8* destI5 = (u8*)destI;
+
+            for (i = pixelTop; i <= pixelBottom; i++) {
+                for (j = pixelLeft; j <= pixelRight; j++) {
+                    pixel = srcBuf[i * bufWidth + j];
+                    r = (pixel >> 11) & 0x1F;
+                    g = (pixel >> 6) & 0x1F;
+                    b = (pixel >> 1) & 0x1F;
+
+                    pixel = 0;
+
+                    *(destI5++) = (((r * 2 + g * 4 + b) * 0xFF) / 217) & 0xF8;
+                }
+            }
+            break;
+        }
+
+        case 8: {
+            u8* destI8 = (u8*)destI;
+
+            for (i = pixelTop; i <= pixelBottom; i++) {
+                for (j = pixelLeft; j <= pixelRight; j++) {
+                    pixel = srcBuf[i * bufWidth + j];
+
+                    r = (pixel >> 11) & 0x1F;
+                    g = (pixel >> 6) & 0x1F;
+                    b = (pixel >> 1) & 0x1F;
+
+                    *(destI8++) = ((r * 2 + g * 4 + b) * 0xFF) / 217;
+                }
+            }
+            break;
+        }
+
+        case 16: {
+            u16* destI16 = (u16*)destI;
+
+            for (i = pixelTop; i <= pixelBottom; i++) {
+                for (j = pixelLeft; j <= pixelRight; j++) {
+                    *(destI16++) = srcBuf[i * bufWidth + j];
+                }
+            }
+            break;
+        }
+    }
+}
+
+void Play_SetMotionBlurPriorityAlpha(u32 alpha) {
+    R_MOTION_BLUR_PRIORITY_ALPHA = alpha;
+}
+
+void Play_EnableMotionBlurPriority(u32 alpha) {
+    R_MOTION_BLUR_PRIORITY_ALPHA = alpha;
+    R_MOTION_BLUR_PRIORITY_ENABLED = true;
+}
+
+void Play_DisableMotionBlurPriority(void) {
+    R_MOTION_BLUR_PRIORITY_ENABLED = false;
+}
+
+typedef enum {
+    /* 0 */ PICTOGRAPH_PHOTO_STATE_OFF,
+    /* 1 */ PICTOGRAPH_PHOTO_STATE_START,
+    /* 2 */ PICTOGRAPH_PHOTO_STATE_TAKE,
+    /* 3 */ PICTOGRAPH_PHOTO_STATE_FINISH
+} PictographPhotoState;
+
+// Will take the photograph, but doesn't compress and save it
+void Play_StartTakingPictographPhoto(void) {
+    R_PICTOGRAPH_PHOTO_STATE = PICTOGRAPH_PHOTO_STATE_START;
+}
+
+void Play_TakePictographPhoto(PreRender* prerender) {
+    PreRender_ApplyFilters(prerender);
+    Play_ConvertBufferToI(gPictoPhotoI8, prerender->fbufSave, 320, 80, 64, 240 - 1, 176 - 1, 8);
+}
+
+s32 Play_ChooseDynamicTransition(PlayState* this, s32 transitionType) {
+    s32 nextTransitionType = transitionType;
+
+    if (transitionType == TRANS_TYPE_FADE_DYNAMIC) {
+        if (!gSaveContext.save.isNight) {
+            nextTransitionType = TRANS_TYPE_FADE_WHITE;
+        } else {
+            nextTransitionType = TRANS_TYPE_FADE_BLACK;
+        }
+    }
```

Comment:
I wonder if this could be called `TRANS_TYPE_FADE_DAYTIME` or something, to indicate that it changes based on day or night?

Reply:
I was confused reading this comment cause "daytime" sounded to me like just "day". I do understand your point now though is "day time based"

I think dynamic is fine though

Reply:
Personally I like the more general dynamic. In theory you could add more dynamic transitions that are not based on the time of day.

Reply:
`TIME_DEPENDENT` is the only alternative I can think of, which is longer but more precise.